packages/peregrine/lib/talons/CartPage/ProductListing/useProduct.js

```diff
@@ -32,6 +32,9 @@ export const useProduct = props => {
             // If a product mutation is in flight, tell the cart.
             setIsCartUpdating(updateItemLoading || removeItemLoading);
         }
+
+        // Reset updating state on unmount
+        return () => setIsCartUpdating(false);
```

Comment:
Should we use the `onCompleted` callback instead? 

https://www.apollographql.com/docs/react/data/mutations/#options